tests/bootstrap_project/test/template_checksum_test.dart

```
@@ -0,0 +1,208 @@
+    try {
+      await Directory(tempPath).delete(recursive: true);
+    } catch (e) {}
+  });
+
+  group('Template checksum verification', () {
+    test(
+        'Given template files when calculating checksums then checksums are correctly generated',
+        () async {
+      var calculateProcess = await runProcess(
+        'bash',
+        ['util/calculate_template_checksum'],
+        workingDirectory: rootPath,
+      );
+
+      expect(
+        calculateProcess.exitCode,
+        0,
+        reason: 'Failed to calculate template checksums',
+      );
+
+      var checksumsFile = File(path.join(templatesPath, 'checksums.json'));
+      expect(checksumsFile.existsSync(), isTrue,
+          reason: 'checksums.json file was not created');
+    });
+
+    test(
+        'Given no cached templates when creating a new project then templates are downloaded and verified',
+        () async {
+      final (:projectName, :commandRoot) = createRandomProjectName(tempPath);
+
+      var serverpodDir = Directory(path.join(
+        Platform.environment['HOME'] ?? Platform.environment['USERPROFILE']!,
+        '.serverpod',
+      ));
+      if (serverpodDir.existsSync()) {
+        try {
+          await serverpodDir.delete(recursive: true);
+        } catch (_) {}
+      }
+
+      var createProcess = await runProcess(
+        'serverpod',
+        ['create', projectName, '--mini', '-v', '--no-analytics'],
+        workingDirectory: tempPath,
+        environment: {
+          'SERVERPOD_HOME': rootPath,
+        },
+      );
+
+      expect(
+        createProcess.exitCode,
+        0,
+        reason: 'Failed to create project with template verification',
+      );
+
+      expect(
+        Directory(path.join(tempPath, projectName)).existsSync(),
+        isTrue,
+        reason: 'Project directory was not created',
+      );
+
+      try {
+        await Directory(path.join(tempPath, projectName))
+            .delete(recursive: true);
+      } catch (_) {}
```

Why not test explicitly for existence instead of squashing all exceptions?